aequilibrae/paths/linear_approximation.py

```diff
@@ -387,7 +387,7 @@ def derivative_of_objective(stepsize):
             # in order to add a small fraction of the AoN. A heuristic value equal to the corresponding MSA step size
             # seems to work well in practice.
             if derivative_of_objective(0.0) < derivative_of_objective(1.0):
-                if self.algorithm == "frank-wolfe":
+                if self.algorithm == "frank-wolfe" or self.stepsize == 0.0:
```

Comment:
I don't think we need to check for zero here because of the preceding line

Reply:
This is EXACTLY what fixes the issue. When the algorithm is NOT FW, but we found a stepsize of zero

Reply:
I am not sure we should special case it, this will make a CFW or BFW step with zero contribution add a little bit (see my TODO on line 396 below). It might be better to just re-start from FW. Have you tested this?

Reply:
Tested what I implemented?  I did it with the exact instance @alexdawn tested with.

What is going on here (for CFW/BFW) is the following:

If the step size is zero (generally when we fail to find a step size):
    Do a FW step:
        if we fail to do compute a step size (we find ourselves in this part of the code by either being FW or due to having a do_fw_step true:
                do MSA step

Does it make sense?

Reply:
There is layer upon layer here, let me think this through before I comment straight off the bat again :)
The step size is only zero here if we set it to zero in the else clause, so this is what the "recursive" below refers to, right? I think this works!

Reply:
However, we would need to throw before setting the step size with the old scipy method on line 379 to be consistent, right?